src/catalog/catalog_set.cpp

```diff
@@ -526,4 +538,29 @@ void CatalogSet::Scan(const std::function<void(CatalogEntry *)> &callback) {
 	}
 }
 
+EntryDropper::EntryDropper(CatalogSet *catalog_set, idx_t entry_index)
+    : catalog_set(catalog_set), entry_index(entry_index) {
+	old_deleted = catalog_set->entries[entry_index].get()->deleted;
+}
+
+EntryDropper::~EntryDropper() {
+	catalog_set->entries[entry_index].get()->deleted = old_deleted;
+}
+
+void EntryDropper::DropEntryDependencies(CatalogSet *catalog_set, ClientContext &context, idx_t entry_index,
```

Comment:
`catalog_set` has to be defined here, right? Could we perhaps move this method to a regular method in CatalogSet, instead of a static method in the EntryDropper class? (i.e. `catalog_set->DropEntryDependencies(context, entry_index, entry, cascade);`)